### ISSUE TITLE
Default subcommand not possible in example

### ISSUE THREAD

**Issue by mfowlewebs** [tj/commander.js#469]
In [Example](https://github.com/tj/commander.js#examples) we have a bunch of subcommands listed, however there is no way to make any of them the default.

I've tried, for example, changing the final subcommand to `.command('*', undefined, {isDefault: true})` however it seems to have no impact. Providing a description does worse: it causes `file-* does not exist` to be returned for either \* or default.

**Comment by subwiz** [comment]
I have this:

```js
program.command('help', null, {isDefault:true})
  .description('display help information.')
  .action(function(){
    program.outputHelp();
  });

program.command('*', null, {noHelp:true})
  .action(function(cmd){
    console.error('unrecognized command: ' + cmd);
    program.outputHelp();
  });
```

`noHelp` is working as expected, while `isDefault` has no impact.

**Comment by steelbrain** [comment]
Bump

**Comment by ivanakimov** [comment]
+1

**Comment by szeck87** [comment]
Is this going to be looked at?  It seems like https://github.com/tj/commander.js/blob/aaa447a8f0717771d0b021c27b709ef867c220e0/index.js#L169-L191 is being called three times.  The first time it correctly finds the isDefault flag, the last two are undefined.

**Comment by padcom** [comment]
Is this package at all maintained anymore? This is really a critical issue and is still not fixed.

**Comment by paulmelnikow** [comment]
These days I'm using:

- https://github.com/nodeca/argparse

And a more popular alternative is:

- https://github.com/yargs/yargs

**Comment by shadowspawn** [comment]
The `isDefault` functionality is currently only supported for git-style subcommands. I am closing this issue in favour of #742 which identifies the limitation in the title.

**Comment by padcom** [comment]
I know this is a lot to ask but could you elaborate (provide a short example maybe?) what does it mean _git-style subcommands_? I am mighty confused...

**Comment by shadowspawn** [comment]
@padcom 
There is a section in the README, and example in the examples:
https://github.com/tj/commander.js#git-style-sub-commands
https://github.com/tj/commander.js/blob/master/examples/pm

**Comment by shadowspawn** [comment]
Support for a default command with action handler shipped in Commander 5.0.0: https://github.com/tj/commander.js/issues/742
